08-components-lab/src/01-timeconverter.jsx

```diff
@@ -8,5 +8,49 @@
 
 import React from 'react';
 
-export default React.createClass({
-});
+export default class TimeConverter extends React.Component{
+    constructor(props){
+        super(props);
+        this.state={
+            number : 0
+        }
+    }
+
+    static propTypes = {
+        number: React.propTypes.number.isRequired
+    };
+
+    getSeconds(){
+        return Number(this.state.number);
+    };
+    getMinutes(){
+        return Number(this.state.number / 60);
+    };
+    getHours(){
+        return Number(this.state.number / 3600);
+    }
+
+    calc = (evt) => {
+        //reset numbers before setting new val
+        this.setState({number:0});
+        if (evt.target.name === "seconds"){
+            this.setState({number:evt.target.value});
+        }
+        else if (evt.target.name === "minutes"){
+            this.setState({number:evt.target.value*60});
+        }
+        else if (evt.target.name === "hours"){
+            this.setState({number:evt.target.value*3600});
+        }
+    };
+
+    render(){
+        return <div>
+            <input type="number" placeholder="enter number of seconds" value={this.getSeconds()} name="seconds" onChange={this.calc} />
+            <br />
```

Comment:
Instead of using `<br/> tag for spacing out input elements it's better using style prop to add inline styling or styling your input fields with css.